crates/arroyo-connectors/src/rabbitmq_stream/mod.rs

```diff
@@ -0,0 +1,276 @@
+            sink: false,
+            testing: false,
+            hidden: false,
+            custom_schemas: true,
+            connection_config: Some(CONFIG_SCHEMA.to_string()),
+            table_config: TABLE_SCHEMA.to_string(),
+        }
+    }
+
+    fn table_type(
+        &self,
+        _: Self::ProfileT,
+        _: Self::TableT,
+    ) -> arroyo_rpc::api_types::connections::ConnectionType {
+        ConnectionType::Source
+    }
+
+    fn test(
+        &self,
+        _: &str,
+        _: Self::ProfileT,
+        _: Self::TableT,
+        _: Option<&arroyo_rpc::api_types::connections::ConnectionSchema>,
+        tx: tokio::sync::mpsc::Sender<arroyo_rpc::api_types::connections::TestSourceMessage>,
+    ) {
+        // TODO
+        tokio::task::spawn(async move {
+            let message = TestSourceMessage {
+                error: false,
+                done: true,
+                message: "Successfully validated connection".to_string(),
+            };
+            tx.send(message).await.unwrap();
+        });
+    }
+
+    fn from_options(
+        &self,
+        name: &str,
+        options: &mut std::collections::HashMap<String, String>,
+        schema: Option<&arroyo_rpc::api_types::connections::ConnectionSchema>,
+        profile: Option<&arroyo_rpc::api_types::connections::ConnectionProfile>,
+    ) -> anyhow::Result<arroyo_operator::connector::Connection> {
+        let connection_config = match profile {
+            Some(connection_profile) => {
+                serde_json::from_value(connection_profile.config.clone())
+                    .map_err(|e| anyhow!("Failed to parse connection config: {:?}", e))?
+            }
+            None => {
+                let host = options.remove("host");
+                let username = options.remove("username");
+                let password = options.remove("password");
+                let virtual_host = options.remove("virtual_host");
+                let port = match options.remove("port") {
+                    Some(v) => Some(v.parse::<u16>()?),
+                    None => None,
+                };
+
+                let tls_config = options
+                    .remove("tls_config.enabled")
+                    .map(|enabled| TlsConfig {
+                        enabled: Some(enabled == "true"),
+                        trust_certificates: options
+                            .remove("tls_config.trust_certificates")
+                            .map(|trust_certificates| trust_certificates == "true"),
+                        root_certificates_path: options.remove("tls_config.root_certificates_path"),
+                        client_certificates_path: options
+                            .remove("tls_config.client_certificates_path"),
+                        client_keys_path: options.remove("tls_config.client_keys_path"),
+                    });
+
+                let load_balancer_mode = options.remove("json.unstructured").map(|t| t == "true");
```

Comment:
I'm not quite sure what's going on here—`json.unstructured` controls our deserialization logic. Was this meant to be another config?

Reply:
Yes, it should be `load_balancer_mode`! Thank you for identifying this during the review process. This option forces the client to connect to the load balancer instead of connecting directly to the node.

https://www.rabbitmq.com/blog/2021/07/23/connecting-to-streams#with-a-load-balancer